src/session.c

```diff
@@ -1006,35 +1006,52 @@ static void increment_node_failure(char *addr, const CURLcode res, const long re
     healthstatus->timestamp = time(NULL); // Most recent failure. We don't currently use this value, but it might be interesting
 }
 
-void process_status(const char *fcn_name, CURL *session, const CURLcode res, 
+bool process_status(const char *fcn_name, CURL *session, const CURLcode res, 
         const long response_code, const long elapsed_time, const int iter, 
         const char *path, bool tmp_session) {
 
+    bool non_retriable_error = false; // default to retry
```

Comment:
It looks like there are only a couple places where we return true.  Would it be better to explicitly return values rather than carry state throughout the function?

Reply:
It just makes it explicit what we are returning.